src/app/courses/courses-router.module.ts

```diff
@@ -14,7 +14,7 @@ import { CoursesEnrollComponent } from './enroll-courses/courses-enroll.componen
 const routes: Routes = [
   { path: '', component: CoursesComponent },
   { path: 'add', component: CoursesAddComponent },
-  { path: 'update/:id', component: CoursesAddComponent },
+  { path: 'view/:id/update', component: CoursesAddComponent },

```

Comment:
Both `update/:id` and `view/:id/update` paths should exist for `CoursesAddComponent`.